core/services/offchainreporting/OffchainReportingAggregator.go

```diff
@@ -0,0 +1,233 @@
+package offchainreporting
+
+import (
+	"context"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/offchainreporting/confighelper"
+	"github.com/smartcontractkit/chainlink/offchainreporting/gethwrappers/offchainaggregator"
+	ocrtypes "github.com/smartcontractkit/chainlink/offchainreporting/types"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/pkg/errors"
+)
+
+const (
+	OffchainReportingAggregatorName = "OffchainReportingAggregator"
+)
+
+var (
+	OffchainReportingAggregatorConfigSet = getConfigSetHash()
+
+	OffchainReportingAggregatorLogTypes = map[gethCommon.Hash]eth.Log{
+		OffchainReportingAggregatorConfigSet: &LogConfigSet{},
+	}
+)
+
+type (
+	LogConfigSet struct {
+		eth.GethRawLog
+		ConfigCount  uint64
+		Oracles      []gethCommon.Address
+		Transmitters []gethCommon.Address
+		Threshold    uint8
+		Config       []byte
+	}
+
+	OffchainReportingAggregator struct {
+		ethClient        eth.Client
+		configSetChan    chan offchainaggregator.OffchainAggregatorConfigSet
+		contractFilterer *offchainaggregator.OffchainAggregatorFilterer
+		contractCaller   *offchainaggregator.OffchainAggregatorCaller
+		contractAddress  gethCommon.Address
+		logBroadcaster   eth.LogBroadcaster
+
+		jobID       models.ID
+		transmitter Transmitter
+		contractABI abi.ABI
+	}
+
+	Transmitter interface {
+		CreateEthTransaction(ctx context.Context, toAddress gethCommon.Address, payload []byte) error
+		FromAddress() gethCommon.Address
+	}
+)
+
+var (
+	_ ocrtypes.ContractConfigTracker      = &OffchainReportingAggregator{}
+	_ ocrtypes.ContractTransmitter        = &OffchainReportingAggregator{}
+	_ ocrtypes.ContractConfigSubscription = &OffchainReportingAggregatorConfigSubscription{}
+)
+
+func NewOffchainReportingAggregator(address gethCommon.Address, ethClient eth.Client, logBroadcaster eth.LogBroadcaster, jobID models.ID, transmitter Transmitter) (o *OffchainReportingAggregator, err error) {
```

Comment:
Mind changing `jobID` to an `int32`?

Reply:
Also, how should I be instantiating `Transmitter` in my PR?

Reply:
If I change `jobID` to `int32` it no longer complies with the interface.

```
Register:
	*OCRContractConfigSubscription does not implement eth.LogListener (wrong type for JobID method)
		have JobID() int32
		want JobID() *models.ID
```

Will have to merge as-is and change it in your PR.

Reply:
You can check `services/offchainreporting/transmitter_test.go` for an example of how to instantiate Transmitter.

Reply:
I guess I'm not understanding why it needs to be a *models.ID.  Shouldn't the Core implementation drive that decision?

Reply:
It doesn't need to be but it breaks the interface if I change it here.